In [5]:
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns
import pandas as pd

In [6]:
df= pd.read_csv("Housing.csv",sep = ',',header = 0,na_values = ["Na","-","null"])

In [7]:
print(df.shape)
print(df.columns)

(545, 13)
Index(['price', 'area', 'bedrooms', 'bathrooms', 'stories', 'mainroad',
       'guestroom', 'basement', 'hotwaterheating', 'airconditioning',
       'parking', 'prefarea', 'furnishingstatus'],
      dtype='object')


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int64 
 1   area              545 non-null    int64 
 2   bedrooms          545 non-null    int64 
 3   bathrooms         545 non-null    int64 
 4   stories           545 non-null    int64 
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int64 
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int64(6), object(7)
memory usage: 55.5+ KB


In [9]:
df.describe()

price          area    bedrooms   bathrooms     stories  \
count  5.450000e+02    545.000000  545.000000  545.000000  545.000000   
mean   4.766729e+06   5150.541284    2.965138    1.286239    1.805505   
std    1.870440e+06   2170.141023    0.738064    0.502470    0.867492   
min    1.750000e+06   1650.000000    1.000000    1.000000    1.000000   
25%    3.430000e+06   3600.000000    2.000000    1.000000    1.000000   
50%    4.340000e+06   4600.000000    3.000000    1.000000    2.000000   
75%    5.740000e+06   6360.000000    3.000000    2.000000    2.000000   
max    1.330000e+07  16200.000000    6.000000    4.000000    4.000000   

          parking  
count  545.000000  
mean     0.693578  
std      0.861586  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      1.000000  
max      3.000000

This table describes how the numeric data is distributed


In [25]:
print(df["mainroad"].value_counts(""))
print(df["guestroom"].value_counts(""))
print(df["basement"].value_counts(""))
print(df["hotwaterheating"].value_counts(""))
print(df["airconditioning"].value_counts(""))
print(df["prefarea"].value_counts(""))
print(df["furnishingstatus"].value_counts(""))

mainroad
yes    468
no      77
Name: count, dtype: int64
guestroom
no     448
yes     97
Name: count, dtype: int64
basement
no     354
yes    191
Name: count, dtype: int64
hotwaterheating
no     520
yes     25
Name: count, dtype: int64
airconditioning
no     373
yes    172
Name: count, dtype: int64
prefarea
no     417
yes    128
Name: count, dtype: int64
furnishingstatus
semi-furnished    227
unfurnished       178
furnished         140
Name: count, dtype: int64


This analysis shows how the categorical features are distributed across the housing dataset.
1) Main road: About 86% of the houses are located on a main road, so this feature is very common and not a strong differentiator.
2) Guest room: Only 18% of houses have a guest room, making it a relatively rare and premium feature.
3) Basement: Around 35% of houses include a basement; most houses do not have one.
4) Hot water heating: Present in only 5% of houses, indicating an extremely rare facility.
5) Air conditioning: Available in approximately 32% of houses, showing moderate adoption.
6) Preferred area: Only 23% of houses are in preferred areas, so such locations are uncommon.
7) Furnishing status:
     42% are semi-furnished
     32% are unfurnished
     26% are fully furnished

From the above analysis, we can conclude that the dataset mainly represents average or middle-range houses rather than luxury houses. Most houses have common features like main road access and are semi-furnished, while facilities such as guest rooms, basements, hot water heating, air conditioning, and preferred area locations are available only in a smaller number of houses.

The categorical data is not evenly distributed, as some categories dominate over others. This shows that certain features are common across most houses, whereas some features are rare and may indicate higher value. Overall, the dataset seems realistic and suitable for further analysis, but care should be taken while modeling because imbalanced categories can affect the results.